### PR TITLE
HARMONY-1597: Scripts and templating to get a quick start on building a service

### PR DESCRIPTION
**Pull request by chris-durbin**
## Jira Issue ID
HARMONY-1597

## Description
The goal of this is to make it easier to get started developing a new harmony service.

## Local Test Steps
Build the harmony docker image to pick up changes that allow local testing without UMM-S.
Run bin/generate-new-service and follow the instructions there.

## PR Acceptance Checklist
* [X] Acceptance criteria met
* [ ] Tests added/updated (if needed) and passing
* [X] Documentation updated (if needed)

### CODE REVIEW

**Comment by vinnyinverso** [comment]
I had a comment (and I think a couple newlines) at the bottom of my .env file, and the script added to it:
`# PODAAC_CONCISE_LIMITS_MEMORY=1000MiTEST_SERV_COLLECTIONS=coll-1,coll-2`

File: bin/generate-new-service [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+${SERVICE_NAME_UPPER}_COLLECTIONS=$COLLECTIONS
+
+EOF
+  sed -i '' "s/^LOCALLY_DEPLOYED_SERVICES=.*/&,${SERVICE_NAME}/" .env
+fi
+
+echo "Successfully setup project. You can take a look at env-defaults, .env, and config/services.yml to see how your service was set up for use in harmony.
+Next steps:
+cd ../harmony-${SERVICE_NAME}
+Modify the .py file to replace the sample code in the process_item method with your service logic.
```
Comment (ygliuvt) [inline bin/generate-new-service:126]:
```suggestion
Modify the ${SERVICE_NAME}.py file to replace the sample code in the process_item method with your service logic.
```

File: docs/guides/adapting-new-services.md [hunk 2]
```diff
@@ -2,6 +2,33 @@
 
 Please reach out in #harmony-service-providers (EOSDIS Slack) for additional guidance on any adaptation needs, and especially with any feedback that can help us improve.
 
+## Quick Start
+Fully setting up a service can be overwhelming for first time service providers. We now provide a script `bin/generate-new-service` to generate
+much of the scaffolding to get services ready for local integration testing with harmony quickly. The scaffolding provides the following:
+
+1. Updates to env-defaults files to add needed environment variables for running the service
+1. Updates to services.yml to fill in a new service definition to call the service
```
Comment (ygliuvt) [inline docs/guides/adapting-new-services.md:10]:
```suggestion
1. Updates to services.yml to fill in a new service definition to call the service in our UAT environment.
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Publish to DockerHub
 on:
   push:
     branches:
-      - main
+      - harmony-1538-publish-fix
```
Comment (indiejames) [inline .github/workflows/publish.yml:6]: I think you need to set this back to `main`

**Review by indiejames — APPROVED** [review]
Tested locally

**Comment by ygliuvt** [comment]
I am able to run the `bin/generate-new-service` script, but harmony fails to come up with `Error: There must be one and only one umm_s record configured as a string for harmony service` error. I guess user is supposed to fill out the umm_s configuration in services.yml? Feels like we are missing some documentation on this front. 

Also the generated umm_s config in the services.yml is: `umm_s: []`, this is misleading people into filling in the value as an array when the expected value is a string.

File: docs/guides/adapting-new-services.md (hunk 2 above)
Comment (chris-durbin) [reply docs/guides/adapting-new-services.md:10]: This quick start isn't meant for running in our UAT environment, it is just for running locally so mentioning calling the service in UAT would be confusing. But you are right though that I need to make it clear somewhere that these changes must be tested with collections in the CMR UAT environment (not production collections) - I'll clarify.

**Comment by chris-durbin** [comment]
> I am able to run the bin/generate-new-service script, but harmony fails to come up with Error: There must be one and only one umm_s record configured as a string for harmony service error.

I forgot to include in the instructions you need to build the harmony image to pick up the code change that allows local testing without a UMM-S record.

**Review by ygliuvt — APPROVED** [review]
Tested successfully locally.

**Pull request merged**